File: 0658-find-k-closest-elements/0658-find-k-closest-elements.py
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        priority_queue<pair<int,int> >heap;
        int n=arr.size();
        for(int i=0;i<n;i++)
        {
            heap.push({abs(x-arr[i]),arr[i]});
            if(heap.size()>k)
            {
                heap.pop();
            }
        }
        vector<int> sol;
        while(heap.size()>0)
        {
            sol.push_back(heap.top().second);
            heap.pop();
        }
        sort(sol.begin(),sol.end());
        return sol;
    }
};

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.js
var findClosestElements = function (arr, k, x) {
  let leftPointer = 0;
  let rightPointer = arr.length - 1;
  while (rightPointer - leftPointer >= k) {
    // a-x <= b-x (Comparison Law at the Question)
    if (Math.abs(arr[leftPointer] - x) <= Math.abs(arr[rightPointer] - x)) rightPointer--;
    else leftPointer++;
  }
  return arr.slice(leftPointer, rightPointer + 1);
};